crates/router/src/core/webhooks/webhook_events.rs

```diff
@@ -22,19 +24,23 @@ enum MerchantAccountOrProfile {
 pub async fn list_initial_delivery_attempts(
     state: SessionState,
     merchant_id: common_utils::id_type::MerchantId,
-    constraints: api::webhook_events::EventListConstraints,
-) -> RouterResponse<Vec<api::webhook_events::EventListItemResponse>> {
-    let profile_id = constraints.profile_id.clone();
-    let constraints =
-        api::webhook_events::EventListConstraintsInternal::foreign_try_from(constraints)?;
+    api_constraints: api::webhook_events::EventListConstraints,
+) -> RouterResponse<api::webhook_events::TotalEventsResponse> {
+    let profile_id = api_constraints.profile_id.clone();
+    let constraints = api::webhook_events::EventListConstraintsInternal::foreign_try_from(
+        api_constraints.clone(),
+    )?;
 
     let store = state.store.as_ref();
     let key_manager_state = &(&state).into();
     let (account, key_store) =
-        get_account_and_key_store(state.clone(), merchant_id, profile_id).await?;
+        get_account_and_key_store(state.clone(), merchant_id.clone(), profile_id.clone()).await?;
+
+    let now = common_utils::date_time::now();
+    let past_90_days_ago = now - time::Duration::days(EVENT_DELIVERY_LIMIT_DAYS);
```

Comment:
We can use a better variable name, that doesn't include the value of the constant itself. This would allow us to update only the constant's value, without changing much elsewhere in code.

Something along the lines of `events_list_begin_time`?

And we can probably use only the date part of "now - 90 days" and then consider time from midnight on that day?
```suggestion
    let events_list_begin_time = (now.date() - time::Duration::days(INITIAL_DELIVERY_ATTEMPTS_LIST_MAX_DAYS)).midnight();
```